graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -484,6 +495,299 @@ auto GetTemporalPickFn(
+    torch::optional<std::vector<int64_t>>& seed_offsets,
+    const std::vector<int64_t>& fanouts, bool return_eids,
+    NumPickFn num_pick_fn, PickFn pick_fn) const {
+  const int64_t num_seeds = seeds.size(0);
+  const auto indptr_options = indptr_.options();
+
+  // Calculate GrainSize for parallel_for.
+  // Set the default grain size to 64.
+  const int64_t grain_size = 64;
+  torch::Tensor picked_eids;
+  torch::Tensor subgraph_indptr;
+  torch::Tensor subgraph_indices;
+  torch::optional<torch::Tensor> subgraph_type_per_edge = torch::nullopt;
+  torch::optional<torch::Tensor> edge_offsets = torch::nullopt;
+
+  bool with_seed_offsets = seed_offsets.has_value();
+  bool hetero_with_seed_offsets = with_seed_offsets && fanouts.size() > 1;
+
+  // Get the number of edge types. If it's homo or if the size of fanouts is 1
+  // (hetero graph but sampled as a homo graph), set num_etypes as 1.
+  // In temporal sampling, this will not be used for now since the logic hasn't
+  // been adopted for temporal sampling.
+  const int64_t num_etypes =
+      (edge_type_to_id_.has_value() && hetero_with_seed_offsets)
+          ? edge_type_to_id_->size()
+          : 1;
+  std::vector<int64_t> etype_id_to_src_ntype_id(num_etypes);
+  std::vector<int64_t> etype_id_to_dst_ntype_id(num_etypes);
+  torch::Tensor subgraph_indptr_substract;
+  // The pick numbers are stored in a single tensor by the order of etype. Each
+  // etype corresponds to a group of seeds whose ntype are the same as the
+  // dst_type. `etype_id_to_num_picked_offset` indicates the beginning offset
+  // where each etype's corresponding seeds' pick numbers are stored in the pick
+  // number tensor.
+  std::vector<int64_t> etype_id_to_num_picked_offset(num_etypes + 1);
+  etype_id_to_num_picked_offset[0] = 0;
+  if (hetero_with_seed_offsets) {
+    for (auto& etype_and_id : edge_type_to_id_.value()) {
+      auto etype = etype_and_id.key();
+      auto id = etype_and_id.value();
+      auto [src_type, dst_type] = utils::parse_src_dst_ntype_from_etype(etype);
+      auto dst_ntype_id = node_type_to_id_->at(dst_type);
+      etype_id_to_src_ntype_id[id] = node_type_to_id_->at(src_type);
+      etype_id_to_dst_ntype_id[id] = dst_ntype_id;
+      etype_id_to_num_picked_offset[id + 1] =
+          seed_offsets->at(dst_ntype_id + 1) - seed_offsets->at(dst_ntype_id) +
+          1;
+    }
+    for (int64_t i = 1; i <= num_etypes; ++i) {
+      etype_id_to_num_picked_offset[i] += etype_id_to_num_picked_offset[i - 1];
+    }
+  } else {
+    etype_id_to_dst_ntype_id[0] = 0;
+    etype_id_to_num_picked_offset[1] = num_seeds + 1;
+  }
+  // `num_rows` indicates the length of `num_picked_neighbors_per_node`, which
+  // is used for storing pick numbers. In non-temporal hetero sampling, it
+  // equals to sum_{etype} #seeds with ntype=dst_type(etype). In homo sampling,
+  // it equals to `num_seeds`.
+  const int64_t num_rows = (fanouts.size() == 1 || !with_seed_offsets)
```

Comment:
The condition here does not seem to be needed as you already initialize `etype_id_to_num_picked_offset[1]` to `num_seeds + 1`.

Reply:
You should be able to simply use `num_rows = etype_id_to_num_picked_offset[num_etypes]`.